sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java

```diff
@@ -481,11 +623,35 @@ void contentFromUrlRunner(Consumer<String> testRunner) {
         testRunner.accept(getStorageTestingFileUrl(FORM_JPG));
     }
 
-    void customFormDataRunner(Consumer<InputStream> testRunner) {
+    void pdfContentFromUrlRunner(Consumer<String> testRunner) {
+        testRunner.accept(getStorageTestingFileUrl(INVOICE_1_PDF));
+    }
+
+    void customFormDataRunner(BiConsumer<InputStream, Long> testRunner) {
+        if (interceptorManager.isPlaybackMode()) {
+            testRunner.accept(new ByteArrayInputStream("testData.png".getBytes()), CUSTOM_FORM_FILE_LENGTH);
+        } else {
+            testRunner.accept(getFileData(getStorageTestingFileUrl(INVOICE_PDF)), CUSTOM_FORM_FILE_LENGTH);
+        }
+    }
+
+    void customFormJpgDataRunner(BiConsumer<InputStream, Long> testRunner) {
+        if (interceptorManager.isPlaybackMode()) {
+            testRunner.accept(new ByteArrayInputStream("testData.png".getBytes()), FORM_1_JPG_FILE_LENGTH);
+        } else {
+            testRunner.accept(getFileData(getStorageTestingFileUrl(FORM_JPG)), FORM_1_JPG_FILE_LENGTH);
+        }
+    }
+
+    void urlDataRunner(Consumer<String> testRunner, String formData) {
```

Comment:
Names seems to be misleading. Is this a urlRunner or a dataRunner?

Reply:
urlRunner